pkg/controller/admissionchecks/provisioning/controller_test.go

```diff
@@ -40,6 +42,25 @@ import (
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
+type fakeControllerRuntimeClient struct {
+	client.Client
+	err error
+}
+
+func (c *fakeControllerRuntimeClient) WithError(err error) {
+	c.err = err
+}
+
+func (c *fakeControllerRuntimeClient) Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error {
+	if c.err != nil {
+		return c.err
+	}
+
+	return c.Client.Create(ctx, obj, opts...)
+}
```

Comment:
Can we use `builder.WithInterceptorFuncs(...)` instead?

Reply:
Done.